### DISCUSSION TITLE
Query logging in middleware

### DISCUSSION THREAD

**Discussion by eomelyusik** [prisma/prisma#12829] [Q&A]
## Problem

it is impossible to reach raw query when login prisma events using middleware:
```javascript
  export type MiddlewareParams = {
    model?: ModelName
    action: PrismaAction
    args: any
    dataPath: string[]
    runInTransaction: boolean
  }
```

## Suggested solution

There is an opportunity to ger raw query using "on query" event, but is it possible to add a raw query to the middleware params?

**Comment by janpio** [comment]
What exactly are you trying to do in the middleware? Are you trying to manipulate the SQL that will be executed? Or add something to that? What problem are you trying to solve?

**Reply by eomelyusik** [reply]
Just logging the raw query

**Reply by janpio** [reply]
The raw query is not available in the middleware, you will need to use the more general logging functionality for that: https://pris.ly/d/logging. Note that it is also only available when it is executed, not before to control the query or similar.

**Comment by francisco-alejandro** [comment]
I have reviewed the event-based logging and I can get the query and its duration. But when I try to implement a request tracing solution that includes Prisma queries information , sometimes the event log is received out the context of the HTTP request (some kind of race condition) and I can not create a span with query information using the instrumentation collector SDK (Instana in this case). However using the middleware I can wrap the query execution with the collector method and get a reliable tracing span. It would be great to have the raw query there, this way Prisma will improve the support to tracing and logging.
Could this be evaluated as a possible feature ?

**Reply by saimonmoore** [reply]
Heavy plus one to this. As far as instrumentation goes it's pretty standard to see the raw sql (would be a huge boost for debugging) because otherwise you need to go through some hoops to get at the generated sql.

**Reply by bravados** [reply]
This would really give a meaningful context to the logging. Could you consider this, prisma devs? 🙏🏼

**Reply by janpio** [reply]
I think you will be happy to find https://github.com/prisma/prisma/issues/9601 then.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
